packages/core/src/components/menu/_common.scss

```diff
@@ -38,18 +40,62 @@ $dark-menu-background-color: $dark-gray4 !default;
   }
 
   .pt-dark & {
-    @include dark-menu-item($disabled-selector);
+    @include dark-menu-item($disabled-selector, $hover-selector);
   }
 }
 
-@mixin dark-menu-item($disabled-selector: ".pt-disabled") {
+@mixin dark-menu-item($disabled-selector: ".pt-disabled", $hover-selector: ":hover") {
   color: inherit;
 
+  &:not(#{$disabled-selector})#{$hover-selector} {
+    background: $dark-menu-item-color-hover;
+    color: inherit;
+  }
+
   &#{$disabled-selector} {
     color: $pt-dark-text-color-disabled;
   }
 }
 
+@mixin menu-item-intent($text-color: $pt-intent-text-colors) {
+  @each $intent, $color in $text-color {
+    &.pt-intent-#{$intent} {
+      color: $color;
+
+      &::before,
+      &::after,
+      .pt-menu-item-label {
+        color: $color;
+      }
+    }
+  }
+
+  @each $intent, $color in $button-intents {
+    &.pt-intent-#{$intent} {
+      &:hover {
+        background: nth(map-get($button-intents, $intent), 1);
+      }
+
+      &:active,
+      &.pt-active {
```

Comment:
Yeah this is tough...
Having different colors messes with our model of `.pt-active` === `:active` elsewhere but for the anticipated usage here, what you propose definitely looks better.